sklearn/naive_bayes.py

```diff
@@ -958,3 +985,245 @@ def _joint_log_likelihood(self, X):
+            n_features is the number of features. Here, each feature of X is
+            assumed to be from a different categorical distribution.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        classes : array-like, shape = [n_classes] (default=None)
+            List of all the classes that can possibly appear in the y vector.
+
+            Must be provided at the first call to partial_fit, can be omitted
+            in subsequent calls.
+
+        sample_weight : array-like, shape = [n_samples], (default=None)
+            Weights applied to individual samples (1. for unweighted).
+
+        Returns
+        -------
+        self : object
+        """
+        self._check_settings()
+        return (super(CategoricalNB, self)
+                .partial_fit(X, y, classes, sample_weight=sample_weight))
+
+    def _check_X(self, X):
+        return check_array(X, accept_sparse=False)
+
+    def _check_X_y(self, X, y):
+        return check_X_y(X, y, accept_sparse=False)
+
+    def _check_settings(self):
+        if self.handle_unknown not in ('ignore', 'warn', 'raise'):
+            raise ValueError("The attribute 'handle_unknown' is '{}' and "
+                             "should be one of 'ignore', 'warn' or 'raise'"
+                             .format(self.handle_unknown))
+
+    def _init_counters(self, n_effective_classes, n_features):
+        self.class_count_ = np.zeros(n_effective_classes, dtype=np.float64)
+        self.category_count_ = [np.zeros((self.class_count_.shape[0], 0))
+                                for _ in range(n_features)]
+        self.feature_cat_index_mapping_ = [{} for _ in range(n_features)]
+
+    def _count(self, X, Y):
+        self.class_count_ += Y.sum(axis=0)
+        for i in range(self.n_features_):
+            X_feature = X[:, i]
+            cats = np.unique(X_feature)
+            self._update_cat_mapping(self.feature_cat_index_mapping_[i], cats)
+            # update category_count_dimensions in case partial_fit is used, to
+            # to account for unseen categories
+            self.category_count_[i] = self._update_cat_count_dims(
+                self.category_count_[i], self.feature_cat_index_mapping_[i])
+            self._update_cat_count(X_feature, Y,
+                                   self.category_count_[i],
+                                   self.feature_cat_index_mapping_[i])
+
+    def _update_cat_mapping(self, cat_mapping, cats):
+        for cat in cats:
+            cat_mapping.setdefault(cat, len(cat_mapping))
+
+    def _update_cat_count_dims(self, cat_count, cat_mapping):
+        diff = len(cat_mapping) - cat_count.shape[1]
+        if diff > 0:
+            return np.pad(cat_count, [(0, 0), (0, diff)], 'constant')
+        return cat_count
+
+    def _update_cat_count(self, X_feature, Y, cat_count, cat_mapping):
+        for j in range(self.class_count_.shape[0]):
+            X_feature_class = X_feature[Y[:, j] > 0]
```

Comment:
Good point, since `Y` is binarized `Y[:, j].astype(bool)` is less confusing and also for performance reasons putting `mask = Y[:, j].astype(bool)` and using it two times surely is faster. Thanks for pointing that out.